### PR TITLE
Migrate from GCM to FCM SDK.

### LINKED ISSUE #842: Google Cloud Messaging deprecated, will be removed no later than April 2019
See the posted info-mail from Google in the issue I opened for openhab-cloud: https://github.com/openhab/openhab-cloud/issues/182

The same should apply for the app, too. As far as I understood it correctly, the client/server can be migrated from GCM to FCM without the need of doing it at the same time, however, GCM can not co-exist with FCM on the same app.

### PR DESCRIPTION
**Pull request by maniac103**
Closes #842

### CODE REVIEW

**Comment by mueller-ma** [comment]
Can you adjust the string `info_openhab_gcm_connected`?

**Comment by maniac103** [comment]
Sure. Do you happen to know why we use outdated Google libraries (12.1, current seems to be 17.x) and/or why dependabot doesn't pick up those?

**Comment by mueller-ma** [comment]
I disabled the auto-updates, because I was afraid it does update to fcm and everything brakes. You can manually bump the dependencies to enable dependabot again.

**Comment by mueller-ma** [comment]
Looks good so far.

**Comment by maniac103** [comment]
The last commits hopefully should resolve that.

File: mobile/build.gradle [hunk 1]
```diff
@@ -127,8 +127,7 @@ dependencies {
         transitive = true
     }
 
-    fullImplementation 'com.google.firebase:firebase-core:12.0.1'
-    fullImplementation 'com.google.firebase:firebase-messaging:12.0.1'
+    fullImplementation 'com.google.firebase:firebase-messaging:17.3.4'
```
Comment (mueller-ma) [inline mobile/build.gradle]: Can you exclude the analytics part? https://github.com/siacs/Conversations/commit/c2067e3b56364317bc384b55e320b72326fb1de6
And add `<meta-data android:name="firebase_analytics_collection_deactivated" android:value="true" />` to the manifest (https://firebase.google.com/support/guides/disable-analytics)?

**Comment by maniac103** [comment]
Did the string update. I left non-English ones untouched; I assume Crowdin will take care of that.

**Comment by cribskip** [comment]
2.6.2-beta crashes on start because of FCM.

java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:309)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process org.openhab.habdroid.beta. Make sure to call FirebaseApp.initializeApp(Context) first.
	at com.google.firebase.FirebaseApp.getInstance(com.google.firebase:firebase-common@@16.0.2:240)
	at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source)
	at org.openhab.habdroid.core.FcmRegistrationService.registerFcm(FcmRegistrationService.java:106)
	at org.openhab.habdroid.core.FcmRegistrationService.onHandleWork(FcmRegistrationService.java:87)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)
	at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
	at android.os.AsyncTask$2.call(AsyncTask.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	... 3 more

**Comment by maniac103** [comment]
@mueller-ma This looks like a different symptom of https://github.com/openhab/openhab-android/issues/794 (see docs at [1]) ... guess we need to do something about that.

[1] https://firebase.google.com/docs/reference/android/com/google/firebase/FirebaseApp

**Comment by mueller-ma** [comment]
> but I might have always built stable.

Me too. I fixed #794 and released a new beta version.

**Pull request merged**